test/test_argo_engine.py

```diff
@@ -504,11 +507,14 @@ def test_argo_engine_get_archived_workflow_log_succeeded():
     engine._get_archived_workflow_details_dict = mock.MagicMock(
         return_value=mock_return_archived_wf
     )
+    engine._get_workflow_node_artifact = mock.MagicMock(
+        return_value="Problem with mutate()"
+    )
     archived_workflow_errors = engine.get_workflow_logs("archived_wf", "archived_uid")
     assert len(archived_workflow_errors) == 1
-    assert archived_workflow_errors[0]["name"] == "step_one_name"
+    assert archived_workflow_errors[0]["pod_type"] == "Retry"
```

Comment:
after reviewing the information more carefully, I think it would be nice to keep "name" there as well. We can always ignore it in the frontend if we don't want to use it anymore. But for now, it seems to contain some useful information like the actual step name (more user-friendly) and the iteration number (e.g. "argo-wrapper-workflow-123456.get-pheno-csv-flow.get-quantitative-pheno-csv(3)" )

Reply:
I didn't remove the `name` for the step. So the name is still kept in the json blob of the Retry node error log. One thing I wanna mention, since we are looking at the **node of Retry type**, the name of the node won't contain the iteration number. The Retry node is like a parent node(e.g.,  "test-error-simulator-pc2x6.error-run-single-assoc"), and each retry will have a name containing iteration number(e.g., "test-error-simulator-pc2x6.error-run-single-assoc(1)") and the type would be "Pod" instead of "Retry"